platforms/core-runtime/logging/src/main/java/org/gradle/internal/deprecation/DeprecationMessageBuilder.java

```diff
@@ -390,13 +393,17 @@ String formatAdvice(List<String> replacements) {
         }
     }
 
+    static Pattern ID_CLEAN_UP_PATTERN = Pattern.compile("[^a-z-]+");
+
     public static String createDefaultDeprecationId(String... ids) {
         // the replaceAll calls are to ensure that the id is a valid identifier.
         // it removes all whitespace and non-alphanumeric characters, and replaces multiple dashes with a single dash.
-        return screamingSnakeToKebabCase(StringUtils.join(ids, "-"))
-            .replaceAll("[^a-z-]", "-")
-            .replaceAll("-+", "-")
-            .replaceAll("^-+|-+$", "");
+        return removeEnd(
```

Comment:
🤔 Since we want optimize constructing a string wouldn't it be faster to:
1. Create a builder
2. iterate over ids. For each id: 
    * find next [A-Za-z] pattern in it, add it to builder (lowercased). Separate by dash.
    * repeat
5. build the string